### PR TITLE
native hessian merge

### PR DESCRIPTION
**Pull request by Leishunyu**
## What is the purpose of the change

https://github.com/apache/incubator-dubbo/issues/3961

## Brief changelog

merge native-thrift to incubator-dubbo

## Verifying this change

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by Leishunyu** [comment]
@ralf0131  can you help me review

**Comment by Leishunyu** [comment]
this pr merge native-thrift and native-hessian

File: dubbo-all/pom.xml [hunk 1]
```diff
@@ -331,6 +338,13 @@
...
+        <dependency>
+            <groupId>org.apache.dubbo</groupId>
+            <artifactId>dubbo-serialization-native-hession</artifactId>
+            <version>${project.version}</version>
+            <scope>compile</scope>
+            <optional>true</optional>
+        </dependency>
         <dependency>
             <groupId>org.apache.dubbo</groupId>
             <artifactId>dubbo-serialization-jdk</artifactId>
```
Comment (beiwei30) [inline dubbo-all/pom.xml]: you forgot this:

```xml
<include>org.apache.dubbo:dubbo-rpc-native-thrift</include>
<include>org.apache.dubbo:dubbo-serialization-native-hession</include>
```

File: dubbo-all/pom.xml [hunk 2]
```diff
@@ -184,6 +184,13 @@
             <scope>compile</scope>
             <optional>true</optional>
         </dependency>
+        <dependency>
+            <groupId>org.apache.dubbo</groupId>
+            <artifactId>dubbo-rpc-native-thrift</artifactId>
```
Comment (beiwei30) [inline dubbo-all/pom.xml]: should we simply replace this into 'dubbo-rpc-thrift'?

File: dubbo-distribution/pom.xml [hunk 3]
```diff
@@ -140,6 +140,11 @@
             <artifactId>dubbo-rpc-thrift</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.dubbo</groupId>
+            <artifactId>dubbo-rpc-native-thrift</artifactId>
+            <version>${project.version}</version>
+        </dependency>
```
Comment (beiwei30) [inline dubbo-distribution/pom.xml]: no 'dubbo-serialization-native-hession'?

File: dubbo-dependencies-bom/pom.xml [hunk 4]
```diff
@@ -106,7 +106,7 @@
         <consul_process_version>2.0.0</consul_process_version>
         <xmemcached_version>1.3.6</xmemcached_version>
         <cxf_version>3.1.15</cxf_version>
-        <thrift_version>0.8.0</thrift_version>
+        <thrift_version>0.11.0</thrift_version>
```
Comment (beiwei30) [inline dubbo-dependencies-bom/pom.xml]: pls. bump up version to the latest. see this: https://github.com/dubbo/rpc-native-thrift-for-apache-dubbo/network/alert/pom.xml/org.apache.thrift:libthrift/open

File: dubbo-distribution/pom.xml (hunk 3 above)
Comment (Leishunyu) [reply dubbo-distribution/pom.xml]: Both are inside

File: dubbo-all/pom.xml [hunk 5]
```diff
@@ -450,13 +457,6 @@
             <scope>compile</scope>
             <optional>true</optional>
         </dependency>
-        <dependency>
-            <groupId>org.apache.dubbo</groupId>
-            <artifactId>dubbo-metadata-report-etcd</artifactId>
```
Comment (beiwei30) [inline dubbo-all/pom.xml]: should not remove this.

File: dubbo-dependencies-bom/pom.xml (hunk 4 above)
Comment (beiwei30) [inline dubbo-dependencies-bom/pom.xml]: should not include thrift.

File: dubbo-rpc/dubbo-rpc-thrift/pom.xml [hunk 6]
```diff
@@ -52,6 +52,7 @@
         <dependency>
             <groupId>org.apache.thrift</groupId>
             <artifactId>libthrift</artifactId>
+            <version>0.8.0</version>
```
Comment (beiwei30) [inline dubbo-rpc/dubbo-rpc-thrift/pom.xml]: should not include thrift.

Comment (Leishunyu) [reply dubbo-rpc/dubbo-rpc-thrift/pom.xml]: ok, i remove it

File: dubbo-all/pom.xml (hunk 5 above)
Comment (Leishunyu) [reply dubbo-all/pom.xml]: get

File: dubbo-dependencies-bom/pom.xml (hunk 4 above)
Comment (Leishunyu) [reply dubbo-dependencies-bom/pom.xml]: get

**Pull request merged**